### PR TITLE
Add thinkmd flavor and option menu item

### PR DESCRIPTION
**Pull request by zzainulabidin**

### CODE REVIEW

File: opensrp-chw/src/main/java/org/smartregister/chw/application/DefaultChwApplicationFlv.java [hunk 1]
```diff
@@ -118,4 +118,9 @@ public boolean showMyCommunityActivityReport() {
     public boolean launchChildClientsAtLogin() {
         return false;
     }
+
+    @Override
+    public boolean useThinkMd() {
+        return true;
```
Comment (githengi) [inline opensrp-chw/src/main/java/org/smartregister/chw/application/DefaultChwApplicationFlv.java:124]: default should be false

File: opensrp-chw/src/main/java/org/smartregister/chw/activity/ChildProfileActivity.java [hunk 2]
```diff
@@ -161,10 +161,14 @@ public boolean onOptionsItemSelected(MenuItem item) {
     @Override
     public boolean onCreateOptionsMenu(Menu menu) {
         super.onCreateOptionsMenu(menu);
-        menu.findItem(R.id.action_sick_child_form).setVisible(ChwApplication.getApplicationFlavor().hasChildSickForm() && flavor.isChildOverTwoMonths(((CoreChildProfilePresenter) presenter).getChildClient()));
+        menu.findItem(R.id.action_sick_child_form).setVisible(ChwApplication.getApplicationFlavor().hasChildSickForm()
+                && flavor.isChildOverTwoMonths(((CoreChildProfilePresenter) presenter).getChildClient())
+                && !ChwApplication.getApplicationFlavor().useThinkMd());
```
Comment (githengi) [inline opensrp-chw/src/main/java/org/smartregister/chw/activity/ChildProfileActivity.java:166]: why not use `flavor` on the line above

File: opensrp-chw/src/main/java/org/smartregister/chw/activity/ChildProfileActivity.java [hunk 3]
```diff
@@ -161,10 +161,14 @@ public boolean onOptionsItemSelected(MenuItem item) {
...
         menu.findItem(R.id.action_sick_child_follow_up).setVisible(false);
         menu.findItem(R.id.action_malaria_diagnosis).setVisible(false);
         menu.findItem(R.id.action_malaria_followup_visit).setVisible(false);
+        menu.findItem(R.id.action_thinkmd_health_assessment).setVisible(ChwApplication.getApplicationFlavor().useThinkMd()
```
Comment (githengi) [inline opensrp-chw/src/main/java/org/smartregister/chw/activity/ChildProfileActivity.java:170]: why not use `flavor` on the line below

File: opensrp-chw/src/main/java/org/smartregister/chw/activity/ChildProfileActivity.java (hunk 2 above)
Comment (zzainulabidin) [reply opensrp-chw/src/main/java/org/smartregister/chw/activity/ChildProfileActivity.java:166]: flavor object is used to provide different implementation for methods, here we only needs to check the thinkmd support, we can check that from ChwApplication.getApplicationFlavor().useThinkMd(). so no need the flavor to use.

File: opensrp-chw/src/main/java/org/smartregister/chw/activity/ChildProfileActivity.java (hunk 3 above)
Comment (zzainulabidin) [reply opensrp-chw/src/main/java/org/smartregister/chw/activity/ChildProfileActivity.java:170]: flavor object is used to provide different implementation for methods, here we only needs to check the thinkmd support, we can check that from ChwApplication.getApplicationFlavor().useThinkMd(). so no need the flavor to use.

File: opensrp-chw/src/main/java/org/smartregister/chw/application/DefaultChwApplicationFlv.java (hunk 1 above)
Comment (zzainulabidin) [reply opensrp-chw/src/main/java/org/smartregister/chw/application/DefaultChwApplicationFlv.java]: Done

**Pull request merged**